app/assets/javascripts/darkswarm/services/customer.js.coffee

```diff
@@ -13,7 +13,17 @@ angular.module("Darkswarm").factory 'Customer', ($resource, Messages) ->
 
   Customer.prototype.update = ->
     @$update().then (response) =>
-      Messages.success(t('js.changes_saved'))
+      if response.gateway_setup_intent_client_secret && $injector.has('stripeObject')
+        stripe = $injector.get('stripeObject')
+        stripe.confirmCardSetup(response.gateway_setup_intent_client_secret).then (result) =>
+          if result.error
+            response.allow_charges = false
```

Comment:
maybe this line deserves a comment or maybe we can extract this line and the following to a method with a good name: this is making a call to the server to disable charges on the customer object if there is a problem authorizing the card, right?

Reply:
That's correct, it's making a server call. What about renaming the variable to clarify what's happening? Like this? e72c7b9c00f6b15b02be7bb4a477e96cc421b709. I can also extract it to a method too if you think that's better.

Reply:
yes, I think extract method is better, it's making a server call. this method is too complicated already.